[docs] add Ray Train API (#7020)
lightgbm-org/LightGBM: docs/Parallel-Learning-Guide.rst

Ray
^^^

> `Ray`_ is a Python-based framework for distributed computing. Ray provides LightGBM support through the Ray Train API with ``LightGBMTrainer``, in addition to the `lightgbm_ray`_ project maintained within the official Ray GitHub organization.

Reviewer: ```suggestion
`Ray`_ is a Python-based framework for distributed computing. Ray provides LightGBM support through the Ray Train API with ``LightGBMTrainer`` and the `lightgbm_ray`_ project maintained within the official Ray GitHub organization.
```

Let's make this language a little more direct.

For the Ray Train API, see `the Ray documentation`_ for usage examples.

For the lightgbm_ray project, see `the lightgbm_ray documentation`_ for usage examples.

.. note::

>   ``ray`` is not maintained by LightGBM's maintainers. Bug reports or feature requests should be directed to https://github.com/ray-project/ray/issues.

Reviewer: Instead of adding another almost-identical sentence, please just change the existing line to being "`lightgbm_ray` and `ray` are not..."
